pgjdbc/src/main/java/org/postgresql/core/PGStream.java

```diff
@@ -226,21 +226,29 @@ public void setMinStreamAvailableCheckDelay(int delay) {
   }
 
   private Socket createSocket(int timeout) throws IOException {
-    Socket socket = socketFactory.createSocket();
-    String localSocketAddress = hostSpec.getLocalSocketAddress();
-    if (localSocketAddress != null) {
-      socket.bind(new InetSocketAddress(InetAddress.getByName(localSocketAddress), 0));
-    }
-    if (!socket.isConnected()) {
-      // When using a SOCKS proxy, the host might not be resolvable locally,
-      // thus we defer resolution until the traffic reaches the proxy. If there
-      // is no proxy, we must resolve the host to an IP to connect the socket.
-      InetSocketAddress address = hostSpec.shouldResolve()
-          ? new InetSocketAddress(hostSpec.getHost(), hostSpec.getPort())
-          : InetSocketAddress.createUnresolved(hostSpec.getHost(), hostSpec.getPort());
-      socket.connect(address, timeout);
-    }
-    return socket;
+    Socket socket = null;
+    try {
+      socket = socketFactory.createSocket();
+      String localSocketAddress = hostSpec.getLocalSocketAddress();
+      if (localSocketAddress != null) {
+        socket.bind(new InetSocketAddress(InetAddress.getByName(localSocketAddress), 0));
+      }
+      if (!socket.isConnected()) {
+        // When using a SOCKS proxy, the host might not be resolvable locally,
+        // thus we defer resolution until the traffic reaches the proxy. If there
+        // is no proxy, we must resolve the host to an IP to connect the socket.
+        InetSocketAddress address = hostSpec.shouldResolve()
+            ? new InetSocketAddress(hostSpec.getHost(), hostSpec.getPort())
+            : InetSocketAddress.createUnresolved(hostSpec.getHost(), hostSpec.getPort());
+        socket.connect(address, timeout);
+      }
+      return socket;
+    } catch ( Exception ex ) {
+      if (socket != null) {
+        socket.close();
```

Comment:
I'm not sure how do we handle it, however, `socket.close()` could throw `IOException`, so we probably want to wrap `socket.close()` with its own `try-catch`, and add the resulting exception to `ex.addSuppressed(..)`

Reply:
Also , it can be a problem with host verification here https://github.com/pgjdbc/pgjdbc/blob/ab0d7c64d619f2b4f5692f1b2d041204ed676333/pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java